cmd/drand-gossip-relay/main.go

```diff
@@ -0,0 +1,230 @@
+			}
+			opts = append(opts, grpc.WithTransportCredentials(creds))
+		} else if cctx.Bool("insecure") {
+			opts = append(opts, grpc.WithInsecure())
+		} else {
+			opts = append(opts, grpc.WithTransportCredentials(credentials.NewTLS(&tls.Config{})))
+		}
+
+		for {
+			conn, err := grpc.Dial(cctx.String("connect"), opts...)
+			if err != nil {
+				log.Warnf("error connecting to grpc: %+v", err)
+				time.Sleep(5 * time.Second)
+			}
+			client := drand.NewPublicClient(conn)
+			err = workRelay(client, t)
+			if err != nil {
+				log.Warnf("error relaying: %+v", err)
+				time.Sleep(5 * time.Second)
+			}
+		}
+
+		return nil
+	},
+}
+
+func workRelay(client drand.PublicClient, t *pubsub.Topic) error {
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+
+	curr, err := client.PublicRand(ctx, &drand.PublicRandRequest{Round: 0})
+	if err != nil {
+		return xerrors.Errorf("getting initial round failed: %w", err)
+	}
+	log.Infof("got latest rand: %d", curr.Round)
+
+	// context.Background() on purpose as this applies to whole, long lived stream
+	stream, err := client.PublicRandStream(context.Background(), &drand.PublicRandRequest{Round: curr.Round})
+	if err != nil {
+		return xerrors.Errorf("getting rand stream: %w", err)
+	}
+
+	for {
+		rand, err := stream.Recv()
+		if err != nil {
+			return xerrors.Errorf("receving on stream: %w", err)
+		}
+
+		randB, err := proto.Marshal(rand)
+		if err != nil {
+			return xerrors.Errorf("marshaling: %w", err)
+		}
+
+		err = t.Publish(context.TODO(), randB)
+		if err != nil {
+			return xerrors.Errorf("publishing on pubsub: %w", err)
+		}
+		log.Infof("Published randomness on pubsub, round: %d", rand.Round)
+	}
+
+}
+
+var clientCmd = &cli.Command{
+	Name: "client",
+	Flags: []cli.Flag{
+		&cli.StringFlag{
+			Name: "connect-to",
+		},
+	},
+	Action: func(cctx *cli.Context) error {
+		priv, _, err := crypto.GenerateEd25519Key(rand.Reader)
+		if err != nil {
+			return xerrors.Errorf("generating ed25519 key: %w", err)
+		}
+
+		h, _, err := lp2p.ConstructHost(priv, "/ip4/0.0.0.0/tcp/0")
+		if err != nil {
+			return xerrors.Errorf("constructing host: %w", err)
+		}
+		ps, err := lp2p.ConstructPubSub(h)
+		if err != nil {
+			return xerrors.Errorf("constructing pubsub: %w", err)
+		}
+
+		addr, err := ma.NewMultiaddr(cctx.String("connect-to"))
+		if err != nil {
+			return xerrors.Errorf("parsing connect-to addr: %w", err)
+		}
+		pi, err := peer.AddrInfoFromP2pAddr(addr)
+		if err != nil {
+			return xerrors.Errorf("constructing peer info: %w", err)
+		}
+		err = h.Connect(context.Background(), *pi)
+		if err != nil {
+			return xerrors.Errorf("could not connect: %w", err)
+		}
+		c, err := client.NewWithPubsub(ps, cctx.String("network-name"))
+		if err != nil {
+			return xerrors.Errorf("constructing client: %w", err)
+		}
+
+		var notifChan <-chan drand.PublicRandResponse
+		var unsub client.UnsubFunc
+		{
+			ch := make(chan drand.PublicRandResponse, 5)
```

Comment:
What is the meaning of 5?

Reply:
Nothing really. Probably anything higher than 2 would be fine, 1 might cause drops under heavy load.